node/src/actors/chain_manager/handlers.rs

```diff
@@ -839,14 +842,39 @@ impl Handler<BuildVtt> for ChainManager {
             Ok(vtt) => {
                 let fut = transaction_factory::sign_transaction(&vtt, vtt.inputs.len())
                     .into_actor(self)
-                    .then(|s, _act, ctx| match s {
+                    .then(|s, act, ctx| match s {
                         Ok(signatures) => {
-                            let transaction =
-                                Transaction::ValueTransfer(VTTransaction::new(vtt, signatures));
-                            let tx_hash = transaction.hash();
-                            ctx.notify(AddTransaction { transaction });
+                            let vtt_tx = VTTransaction::new(vtt, signatures);
+
+                            let current_epoch = act.current_epoch.unwrap();
+                            let epoch_constants = act.epoch_constants.unwrap();
+                            let mut signatures_to_verify = vec![];
+                            let block_number = act.chain_state.block_number();
+                            let utxo_diff =
+                                UtxoDiff::new(&act.chain_state.unspent_outputs_pool, block_number);
+
+                            match validate_vt_transaction(
+                                &vtt_tx,
+                                &utxo_diff,
+                                current_epoch,
+                                epoch_constants,
+                                &mut signatures_to_verify,
+                                act.consensus_constants().max_vt_weight,
+                            ) {
+                                Ok(_) => {
+                                    let transaction = Transaction::ValueTransfer(vtt_tx);
+                                    let tx_hash = transaction.hash();
 
-                            actix::fut::ok(tx_hash)
+                                    ctx.notify(AddTransaction { transaction });
```

Comment:
Wouldn't it be better to use .send here? The validation is already done in AddTransaction, this way we will validate twice.